Components

christianonoh opened the pull request:
This pull request creates a new Calculator component and removes all the boilerplate from the Create React App. The Calculator component implements the design provided in the task, using functional components and CSS. It also includes a child component for the display area.

To achieve this, I performed the following steps:

- Removed all the boilerplate from the Create React App (text, images, styles).
- Created a new directory called components.
- Inside components, created a new Calculator.js file.
- Implemented the design of the calculator in Calculator.js using JSX and CSS.
- Created a child component for the display area.
- Exported the Calculator component from Calculator.js.
- Imported and used the Calculator component in the App.js top-level component.
- Tested the app in the browser, and it renders without any issues.

christianonoh committed:
Add prop-type

christianonoh committed:
Add Button component

christianonoh committed:
Add Calculator component

christianonoh committed:
Add styling

christianonoh committed:
Modify App.js

christianonoh committed:
Update README file

Shedrack-Sunday commented on src/App.css:
```diff
@@ -1,39 +1,48 @@
-  justify-content: center;
-  font-size: calc(10px + 2vmin);
-  color: white;
+.screen {
+  height: 150px;
+  background-color: aquamarine;
+}
+
+.buttons {
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
+  background-color: black;
+  height: 30rem;
+  gap: 0.8rem;
+  padding: 1rem;
```
- [ ] Firstly, i must admit you have a very beautiful design 👏 . However, the requirement was to stick to the design and have a replication. you can find the instruction below.
- the idea is to see your skill in exact replication based on a given design. you'll have space to have your personification later 

## Requirement:
![image](https://user-images.githubusercontent.com/50990925/234450219-1fbd3ff6-19db-4c10-ba0c-0b1acf5d3ac7.png)

# Your design:
![image](https://user-images.githubusercontent.com/50990925/234450043-f1189e17-ceae-498a-96ce-22eb45a2a824.png)

## Design given:
![image](https://user-images.githubusercontent.com/50990925/234450092-56048fac-81f3-4585-af9f-dbc4aa065228.png)

Shedrack-Sunday reviewed (changes requested):
Hello @christianonoh      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 
![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)

## Highlights 
- Nice project setup✔️  
- Linters rightly setup ✔️
- Nice use of functional component ✔️  
  
# Required Changes  🌵 

- [ ] I see you did a great work on having your own personal design, but check out my review on the project requirements 

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

christianonoh committed:
Fix errors

HAMMAS-SALEEM commented on src/App.css:
```diff
@@ -13,21 +13,23 @@ body {
 
```
### Your Design
![image](https://user-images.githubusercontent.com/93039359/234454307-9b9c20b4-b0ef-4fd2-8196-49b01608f99a.png)

### Expected Design
![image](https://user-images.githubusercontent.com/93039359/234454359-67ebeac3-4fa1-413a-ae4d-34a9b5116f84.png)

- [ ] It would be nice if you follow the design to make your project look attractive and professional as suggested by the previous reviewer :+1: because right now I have seen a lot of things which still need more attention :fearful:
* First the calculator should have more width than height as you can see in the expected design
* Second I saw a black border line around buttons which is not really visible in the expected design.
* Last, The screen of the calculator should have same width, height and the zero **(0)** should be on the right side not on the left side of the screen :+1:

Kindly make all these changes to make your project look professional :blush:

HAMMAS-SALEEM replied:
![image](https://user-images.githubusercontent.com/93039359/234454982-a2f7b3dc-d267-45a7-ab53-27637ff3a801.png)

**[OPTIOANAL]**

- [ ] Kindly remove this live demo section since you don't have any live link for your application :thinking: Once you would have a live link then you can add this live demo section again in your readme to make your readme professional and more useful :+1:

HAMMAS-SALEEM reviewed (changes requested):
# Required Changes :recycle:
![almost-there](https://user-images.githubusercontent.com/65860678/170491251-b1add8ba-b18f-4648-a893-20e510d957fd.gif)

## Hey @christianonoh :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requested changes, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

christianonoh committed:
Fix error

AlphaNtihinduka commented on src/App.css:
```diff
@@ -1,39 +1,48 @@
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
 }
 
-.App-logo {
-  height: 40vmin;
-  pointer-events: none;
+body {
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
 }
```
- [ ] I understand that working on the design may be frustrating and I congratulate you to have a very nice design. However, as the previous reviewers suggest, kindly try as much as possible to stick to the provided calculator design. You should atleast make sure that the zero is aligned to the center right of the calculator to match the original design.
- [ ] In addition, kindly make sure that the width is relatively larger than the height. This will help us to make a professionally designed calculator.
![image](https://user-images.githubusercontent.com/99394374/234542720-f7c9009a-6d04-410d-aaf3-aa8b967fbc4e.png)

AlphaNtihinduka reviewed (changes requested):
## Hi @christianonoh 🙋‍♂️
Good job working on implementing the changes required by the previous reviewers. You have done the design to do the expectation and incredibly well. 👏  

TO HIGHLIGHT:
- Your project is professional ✅
- Professional README file :100: 
- Your PR is not yet professional ✅
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

![close](https://user-images.githubusercontent.com/99394374/210771148-619a4265-33b2-4e68-8901-6bb52693d6de.gif)

## Required changes :recycle: 
- Kindly consider the inline review comments.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

ahmetbozaci commented:
### Hello @christianonoh 
### _This is an additional review as a result of the second opinion request._ 

- For this milestone, it is important that your calculator design closely aligns with the design provided in the requirements. While it may not need to be an exact replica, it should closely resemble the wireframe provided.



___
_**Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away**_

**_As this was the only reason your project was not approved it will be approved as soon as possible._**



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

christianonoh commented:
@ahmetbozaci 
Firstly, I would like to express my gratitude for taking the time to review my work thoroughly. After reviewing your feedback and implementing the changes, I believe I have addressed all the issues you have highlighted.

However, to ensure that I have made all the required changes, I would like to kindly request your clarification on whether or not I have missed any necessary modifications. I appreciate your expertise and feedback, and I am committed to delivering high-quality work that meets the project's requirements.

danmainah reviewed (approved):
Hi @christianonoh , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ahmetbozaci commented:
> @ahmetbozaci Firstly, I would like to express my gratitude for taking the time to review my work thoroughly. After reviewing your feedback and implementing the changes, I believe I have addressed all the issues you have highlighted.
> 
> However, to ensure that I have made all the required changes, I would like to kindly request your clarification on whether or not I have missed any necessary modifications. I appreciate your expertise and feedback, and I am committed to delivering high-quality work that meets the project's requirements.

@christianonoh 
_While it may not need to be an exact replica, it should closely resemble the wireframe provided._ Sorry ı forget to add this: Your design already closely resembles the wireframe provided, so no need to change anything.
